test.js

```diff
@@ -26,6 +26,57 @@ test('get', t => {
 	t.true(dotProp.get({'foo\\.bar': true}, 'foo\\\\.bar'));
 	t.is(dotProp.get({foo: 1}, 'foo.bar'), undefined);
 
+	t.true(dotProp.get([true, false, false], '[0]'));
+	t.true(dotProp.get([{foo: [true]}], '[0].foo[0]'));
+	t.true(dotProp.get({foo: [0, {bar: true}]}, 'foo[1].bar'));
+
+	t.false(dotProp.get(['a', 'b', 'c'], '3', false));
+	t.false(dotProp.get([{foo: [1]}], '[0].bar[0]', false));
+	t.false(dotProp.get([{foo: [1]}], '[0].foo[1]', false));
+	t.false(dotProp.get({foo: [0, {bar: 2}]}, 'foo[0].bar', false));
+	t.false(dotProp.get({foo: [0, {bar: 2}]}, 'foo[2].bar', false));
+	t.false(dotProp.get({foo: [0, {bar: 2}]}, 'foo[1].biz', false));
+	t.false(dotProp.get({foo: [0, {bar: 2}]}, 'bar[0].bar', false));
```

Comment:
Can you add a bunch of tests that tries to trick the parsing? With escapes in different places and stuff. I'm quite paranoid that this change will introduce some kind of vulnerability.